packages/mgt-components/src/graph/graph.user.ts

```diff
@@ -80,23 +94,37 @@ export async function getMe(graph: IGraph): Promise<User> {
  * @returns {(Promise<User>)}
  * @memberof Graph
  */
-export async function getUser(graph: IGraph, userPrincipleName: string): Promise<User> {
+export async function getUser(graph: IGraph, userPrincipleName: string, requestedProps?: string[]): Promise<User> {
   const scopes = 'user.readbasic.all';
   let cache: CacheStore<CacheUser>;
 
   if (getIsUsersCacheEnabled()) {
     cache = CacheService.getCache<CacheUser>(schemas.users, schemas.users.stores.users);
     // check cache
     const user = await cache.getValue(userPrincipleName);
+    let uniqueProps;
+    if (user.user) {
+      const cachedData = JSON.parse(user.user);
+      uniqueProps = requestedProps.filter(prop => !Object.keys(cachedData).includes(prop));
+    }
+
     // is it stored and is timestamp good?
     if (user && getUserInvalidationTime() > Date.now() - user.timeCached) {
       // return without any worries
-      return user.user ? JSON.parse(user.user) : null;
+      if (uniqueProps.length <= 1) {
+        return user.user ? JSON.parse(user.user) : null;
+      }
     }
   }
+
+  let apiString = '/users/${userPrincipleName}';
```

Comment:
Make sure to use backticks, otherwise the string will be a literal

```suggestion
  let apiString = `/users/${userPrincipleName}`;
```

Reply:
backticks added, tested upn type:
a003c5a